detectors/node/opentelemetry-resource-detector-aws/test/detectors/AwsBeanstalkDetector.test.ts

```diff
@@ -105,13 +108,14 @@ describe('BeanstalkResourceDetector', () => {
 
   it('should return empty resource when config file does not exist', async () => {
     fileStub = sinon
-      .stub(AwsBeanstalkDetector, 'fileAccessAsync' as any)
+      .stub(AwsBeanstalkDetectorSync, 'fileAccessAsync' as any)
       .rejects(err);
     readStub = sinon
-      .stub(AwsBeanstalkDetector, 'readFileAsync' as any)
+      .stub(AwsBeanstalkDetectorSync, 'readFileAsync' as any)
       .resolves(JSON.stringify(data));
 
     const resource = await awsBeanstalkDetector.detect();
+    await resource.waitForAsyncAttributes?.();
```

Comment:
note for reviewer: now `awsBeanstalkDetector.detect()` resolves to a promise before the async attributes are resolved so we need to await